filesystem/etc/calico/confd/templates/bird.cfg.template

```diff
@@ -59,43 +59,120 @@ template bgp bgp_template {
   graceful restart;  # See comment in kernel section about graceful restart.
 }
 
-# ------------- Node-to-node mesh -------------
-{{if (json (getv "/global/node_mesh")).enabled}}
-{{range $host := lsdir "/host"}}
-{{$onode_as_key := printf "/host/%s/as_num" .}}
-{{$onode_ip_key := printf "/host/%s/ip_addr_v4" .}}{{if exists $onode_ip_key}}{{$onode_ip := getv $onode_ip_key}}
-{{$nums := split $onode_ip "."}}{{$id := join $nums "_"}}
+# -------------------------------------------------
+# ------------- Node-to-node mesh -----------------
+{{- $node_cid_key := printf "/host/%s/cluster_id" (getenv "NODENAME")}}
+{{- $node_cluster_id := getv $node_cid_key}}
+{{- if ne "" ($node_cluster_id)}}
+
+# This node is configured as a RR in $node_cluster_id. Ignore node-to-node mesh setting.
+{{- else}}
+{{- if (json (getv "/global/node_mesh")).enabled}}
+{{- range $host := lsdir "/host"}}
+{{- $onode_as_key := printf "/host/%s/as_num" .}}
+{{- $onode_ip_key := printf "/host/%s/ip_addr_v4" .}}{{if exists $onode_ip_key}}{{$onode_ip := getv $onode_ip_key}}
+{{- $nums := split $onode_ip "."}}{{$id := join $nums "_"}}
+
 # For peer {{$onode_ip_key}}
-{{if eq $onode_ip ($node_ip) }}# Skipping ourselves ({{$node_ip}})
-{{else if ne "" $onode_ip}}protocol bgp Mesh_{{$id}} from bgp_template {
+{{- if eq $onode_ip ($node_ip) }}
+# Skipping ourselves ({{$node_ip}})
+{{- else if ne "" $onode_ip}}
+protocol bgp Mesh_{{$id}} from bgp_template {
   neighbor {{$onode_ip}} as {{if exists $onode_as_key}}{{getv $onode_as_key}}{{else}}{{getv "/global/as_num"}}{{end}};
-}{{end}}{{end}}{{end}}
-{{else}}
+}
+{{- end}}{{end}}{{end}}
+{{- else}}
 # Node-to-node mesh disabled
-{{end}}
+{{- end}}
+{{- end}}
+
+# -------------------------------------------------
+# ------------ Configured peers -------------------
+
+{{- if ls "/global/peer_sel"}}
+{{- range gets "/global/peer_sel/*"}}{{$data := json .Value}}
+  {{- if selectsme $data.node_selector}}
+
+# BGPPeer {{$data.name}}
+# {{$data}}
 
+    {{- if ne "" $data.ip}}
 
-# ------------- Global peers -------------
-{{if ls "/global/peer_v4"}}
-{{range gets "/global/peer_v4/*"}}{{$data := json .Value}}
-{{$nums := split $data.ip "."}}{{$id := join $nums "_"}}
-# For peer {{.Key}}
-protocol bgp Global_{{$id}} from bgp_template {
+# Peer with IP {{$data.ip}}
+      {{- $nums := split $data.ip "."}}{{$id := join $nums "_"}}
+      {{- $node_cid_key := printf "/host/%s/cluster_id" (getenv "NODENAME")}}{{if exists $node_cid_key}}{{$cluster_id := getv $node_cid_key}}
```

Comment:
There are a bunch of places in this config where we re-query the cluster_id - I don't think this is needed any more, since we store it off in a variable called `node_cluster_id`. Might be able to clean that up a bit.